### PR TITLE
[11.x] Add Databases nightly workflow

### PR DESCRIPTION
**Pull request by Jubeki**
Alternative to https://github.com/laravel/framework/pull/51193

This PR adds a database nightly workflow, which runs preview versions of databases (currently only MariaDB) to make sure upcoming versions are supported.

Note: The line were it runs on pull_requests needs to be removed before merge. It is there only too make sure the tests pass here.

### CODE REVIEW

File: .github/workflows/databases-nightly.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: databases-nightly
+
+on:
+  pull_request: # TODO: Remove before merge
```
Comment (Jubeki) [inline .github/workflows/databases-nightly.yml:4]:
```suggestion
```

Should be deleted before merge.

File: .github/workflows/databases-nightly.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 8.2
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, :php-psr
+          tools: composer:v2
+          coverage: none
+
+      - name: Set Framework version
+        run: composer config version "11.x-dev"
```
Comment (driesvints) [inline .github/workflows/databases-nightly.yml:41]: Why is this needed?

Comment (Jubeki) [reply .github/workflows/databases-nightly.yml:41]: I copied that from the other workflows. Not sure if it is needed.

Comment (driesvints) [reply .github/workflows/databases-nightly.yml:41]: Ah I remember now. All good.

File: .github/workflows/databases-nightly.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+on:
+  pull_request: # TODO: Remove before merge
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+
+ mariadb:
+    runs-on: ubuntu-22.04
```
Comment (crynobone) [inline .github/workflows/databases-nightly.yml:11]: indentation seem's wrong here? 2 space for each indent, right?

Comment (Jubeki) [reply .github/workflows/databases-nightly.yml:11]: Thanks. I fixed it. (Github Actions has still run the job interestingly)

File: .github/workflows/databases-nightly.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+on:
+  pull_request: # TODO: Remove before merge
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+
+  mariadb:
+    runs-on: ubuntu-22.04
```
Comment (crynobone) [inline .github/workflows/databases-nightly.yml:11]:
```suggestion
    runs-on: ubuntu-22.04
    continue-on-error: true
```

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idcontinue-on-error

**Review by crynobone — CHANGES_REQUESTED** [review]

**Review by crynobone — APPROVED** [review]

**Comment by driesvints** [comment]
Thanks @Jubeki!

File: .github/workflows/databases-nightly.yml (hunk 1 above)
Comment (driesvints) [reply .github/workflows/databases-nightly.yml:4]: Of course I forgot this 😂 

Removed now

Comment (Jubeki) [reply .github/workflows/databases-nightly.yml:4]: Thanks! I just wanted to create a PR, because I also just thought I probably forgot to remove that.
